### DISCUSSION TITLE
Possible `vertico-initial-inputs-alist`?

### DISCUSSION THREAD

**Discussion by leungbk** [minad/vertico#422] [Ideas]
`ivy.el` defines the following:

```elisp
(defcustom ivy-initial-inputs-alist
  '((org-refile . "^")
    (org-agenda-refile . "^")
    (org-capture-refile . "^")
    (Man-completion-table . "^")
    (woman . "^"))
  "An alist associating commands with their initial input.

Each cdr is either a string or a function called in the context
of a call to `ivy-read'."
  :type '(alist
          :key-type (symbol)
          :value-type (choice (string) (function))))
```

I am aware that `consult` commands can be customized in this manner via `(consult-customize consult-man :initial "^")`; however, I'm not aware of a way of doing this for any command for which a `consult` wrapper has not been defined.

Would it make sense to expose a similar `defcustom` here in `vertico`?

**Comment by minad** [comment]
> Would it make sense to expose a similar defcustom here in vertico?

I don't plan to add such a customization variable here, but you can use `vertico-multiform-mode` and use a function which inserts after minibuffer setup, see https://github.com/minad/vertico?tab=readme-ov-file#configure-vertico-per-command-or-completion-category. The READMe has this example which you can adapt (insert instead of text-scale-set in the lambda):

~~~ elisp
;; Configure `consult-outline' as a scaled down TOC in a separate buffer
(setq vertico-multiform-commands
      `((consult-outline buffer ,(lambda (_) (text-scale-set -1)))))
~~~

The lambda can apply arbitrary customizations of the command-specific minibuffer.
